modules/game/src/main/Game.scala

```diff
@@ -111,12 +112,28 @@ case class Game(
 
   def lastMoveTimeInSeconds: Option[Int] = lastMoveTime.map(x => (x / 10).toInt)
 
-  def moveTimes(color: Color): Option[List[FiniteDuration]] = {
-    val pivot = if (color == startColor) 0 else 1
-    clockHistory.moveTimes(playedTurns).map(_.toList.zipWithIndex.collect {
-      case (e, i) if (i % 2) == pivot => e
-    })
-  }
+  def moveTimes: Option[Vector[FiniteDuration]] =
+    for {
+      a <- moveTimes(startColor)
+      b <- moveTimes(!startColor)
+    } yield a.zipAll(b, 0.millis, 0.millis).map { case (x, y) => List(x, y) }.flatten.take(playedTurns).toVector
+
+  def moveTimes(color: Color): Option[List[FiniteDuration]] =
+    clockHistory.flatMap { history =>
+      clock.map { clk =>
+        val clockTimes = history(color) :+ (clk.remainingTime(color) * 1000).toLong.millis
```

Comment:
maybe add a `remainingDuration(color: Color): FiniteDuration` function to scalachess' Clock?